"Registering SwissLipids identifiers in Wikidata"

egonw opened the pull request:
per requested port.

egonw committed:
Initial port of https://github.com/elixir-europe/biohackathon-projects-2021/blob/main/projects/6/SwissLipid-in-Wikidata.md

egonw committed:
No references yet

egonw committed:
Added missing ToC entry

egonw committed:
Added obligatory comment: objectives and requirements

egonw committed:
First reference

egonw committed:
More context

egonw committed:
Added the missing journal

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+
+## Main Objectives
+
+The objective of this recipe is to describe how external identifiers can be added to Wikidata in an automated.
```
```suggestion
The objective of this recipe is to describe how [Wikidata](https://www.wikidata.org) can be augmented by adding `external identifiers`, that is identifiers minted by other authorities for an entity described in Wikidata,  in an automated way. This is to make the content more findable and more interoperable.
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+
+## Requirements
+
+To record identifiers of some resource to make the content more findable and more interoperable, first a [Wikidata](https://www.wikidata.org/) property
+needs to be requested. That process is described [here](https://www.wikidata.org/wiki/Wikidata:Property_proposal).
```
```suggestion
- Familiarity with the process of requesting a new property from [Wikidata](https://www.wikidata.org/).
That process is described [here](https://www.wikidata.org/wiki/Wikidata:Property_proposal).
- In order to record identifiers of some resource, first a  property needs to be requested.
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+
+## Main Content
+
+Making datasets more findable and databases more interoperable, it is recommended to share not just top-level metadata, but also record-level
```
a bit unclear as it leads to the question "what is top-level metadata and what is record-level metadata? ". I would suggest rephrasing or clarifying what you mean here.

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+in your own database. That allows other databases, like Wikidata, to link to your database. First, that makes your data more findable,
+but it also makes your data easier to integrate and therefore more interoperable.
+
+This recipe follows this idea and starts with a CCZero-licensed mapping file, linking InChIKeys and database identifiers.
```
```suggestion
This recipe follows this idea and starts with a mapping file that links InChIKeys and database identifiers and which is shared under a [CC0 license](https://creativecommons.org/share-your-work/public-domain/cc0/).
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+but it also makes your data easier to integrate and therefore more interoperable.
+
+This recipe follows this idea and starts with a CCZero-licensed mapping file, linking InChIKeys and database identifiers.
+Recipe *InChI and SMILES identifiers for chemical structures* ([fcb:FCB007](https://w3id.org/faircookbook/FCB007)) explains how InChIKeys
```
```suggestion
The FAIR cookbook Recipe *[InChI and SMILES identifiers for chemical structures (fcb:FCB007)](https://w3id.org/faircookbook/FCB007)*  explains how InChIKeys
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+Recipe *InChI and SMILES identifiers for chemical structures* ([fcb:FCB007](https://w3id.org/faircookbook/FCB007)) explains how InChIKeys
+can be generated for chemical compounds in your database.
+
+As an example, this recipe demonstrates the approach of listing [SwissLipid](http://www.swisslipids.org/#/) {footcite}`Aimo2015SwissLipids`
```
```suggestion
To illustrate our point, this recipe demonstrates the approach of listing [SwissLipid](http://www.swisslipids.org/#/) {footcite}`Aimo2015SwissLipids`
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+can be generated for chemical compounds in your database.
+
+As an example, this recipe demonstrates the approach of listing [SwissLipid](http://www.swisslipids.org/#/) {footcite}`Aimo2015SwissLipids`
+identifiers in Wikidata, developed at the 2021 BioHackathon Europe. Earlier, a [Swiss Lipid property for Wikidata](https://www.wikidata.org/wiki/Wikidata:Property_proposal/SwissLipids_identifier)
```
```suggestion
identifiers in Wikidata, developed at the 2021 BioHackathon Europe.
The work relied on the fact that a [`Swiss Lipids property` for Wikidata](https://www.wikidata.org/wiki/Wikidata:Property_proposal/SwissLipids_identifier)
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+As an example, this recipe demonstrates the approach of listing [SwissLipid](http://www.swisslipids.org/#/) {footcite}`Aimo2015SwissLipids`
+identifiers in Wikidata, developed at the 2021 BioHackathon Europe. Earlier, a [Swiss Lipid property for Wikidata](https://www.wikidata.org/wiki/Wikidata:Property_proposal/SwissLipids_identifier)
+was already proposed, later approved, and created in Wikidata [just before the 2020 BioHackathon Europe](https://www.wikidata.org/w/index.php?title=Property:P8691&oldid=1287579005).
+An existing Wikidata property is a requirement to adding the external identifiers.
```
````suggestion

```{warning}
An existing Wikidata property is a requirement to adding the external identifiers.
```

```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+was already proposed, later approved, and created in Wikidata [just before the 2020 BioHackathon Europe](https://www.wikidata.org/w/index.php?title=Property:P8691&oldid=1287579005).
+An existing Wikidata property is a requirement to adding the external identifiers.
+
+The recipe uses [Bacting](https://github.com/egonw/bacting) {footcite}`Willighagen2021` which can be used in Groovy and Python. We
```
@egon: Bacting isn't available from biotools. Would you be able to submit and link back to this recipe?

egonw replied:
Good question. I need to check. In the past, bio.tools was pretty much about services, not software. Did this change?

egonw replied:
Okay, I misremembered that. They indeed support just software fully too. I'll make an entry!

egonw replied:
https://bio.tools/bacting

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+An existing Wikidata property is a requirement to adding the external identifiers.
+
+The recipe uses [Bacting](https://github.com/egonw/bacting) {footcite}`Willighagen2021` which can be used in Groovy and Python. We
+will use the first here. The starting point is mappings (tuples) that link the SwissLipid identifiers to InChIKeys. The latter
```
```suggestion
will use the first here.

The starting point is mappings (tuples) that link the `SwissLipids` identifiers to `InChIKeys`.
The latter
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+
+### Step 1: getting the data
+
+The data is CC-BY but it was agreed that adding the SwissLipid identifiers to Wikidata (CCZero) is okay.
```
```suggestion
While the SwissLipids data is available under [CC-BY license](https://creativecommons.org/licenses/by/4.0/), it was agreed that adding the SwissLipids identifiers to Wikidata resource, the data of which is available under [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/) is legitimate.
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+
+### Step 2: extract Swiss Lipid ID <> InChIKey tuples
+
+For this step I use `csvtool` (`apt get install csvtool`):
```
```suggestion
For this step, I use `csvtool` (`apt get install csvtool`):
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
````diff
@@ -0,0 +1,162 @@
+   777957 swisslipids.tsv
+```
+
+### Step 3: creating a ShEx model
````
@egonw:
may be add a statement regarding the selection of shex over a W3C standard (shacl) (even if converters exist).

egonw replied:
Sorry I missed this, but this step is actually still empty. I will remove it for now.

egonw replied:
Added SHACL and updated the text a bit.

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+
+### Step 4: creating QuickStatements
+
+Now we have the mappings and the data model in Wikidata, we can create QuickStatements to allow us to enter the
```
```suggestion
Now we have the mappings and the data model in Wikidata, we can create [QuickStatements](https://www.wikidata.org/wiki/Help:QuickStatements) to allow us to enter the
```

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+QuickStatements: [https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy](https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy) This script is using [Apache Groovy](http://www.groovy-lang.org/)
+but Bacting can also be using in Python, see [https://github.com/cthoyt/pybacting](https://github.com/cthoyt/pybacting).
+
+This script uses a federated query against https://beta.sparql.swisslipids.org/sparql/ after a suggestion by Dr [Jerven Bolleman](https://orcid.org/0000-0002-7449-1266)
```
@egonw -> add Jergen to the list of authors/contributors.

egonw replied:
It doesn't feel right to have him under the section "Authors". I will add an Acks section, as I want to ack the ELIXIR BioHackathon and SwissLipid too.

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+
+Based on existing Bacting scripts, a script is created to take the `swisslipids_ids.tsv` file as input and create
+QuickStatements: [https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy](https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy) This script is using [Apache Groovy](http://www.groovy-lang.org/)
+but Bacting can also be using in Python, see [https://github.com/cthoyt/pybacting](https://github.com/cthoyt/pybacting).
```
@egonw : what about adding a jupyter notebook using pybacting?

egonw replied:
yes, I like that idea. I had to get more used to this but found my way around now, including booting a notebook in Google collab: https://bioclipse.github.io/bioclipse.scripting/code/ParseIUPACName.code.html

egonw replied:
I will do that in an update. I was thinking of doing this for https://faircookbook.elixir-europe.org/content/recipes/infrastructure/chemical-identities.html too

proccaserra commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,162 @@
+
+## Conclusion
+
+It will be clear Bacting can be replaced without too much effort by other tools making the same steps. What is the
+main conclusion is that we can indeed automate adding external identifiers routinely to Wikidata using InChIKey
+matching.
+
```
```suggestion
The main conclusion is that it is possible to automate adding external / third party identifiers routinely to Wikidata using InChIKey matching.
The tool `Bacting` used in the example can obviously be replaced without too much effort by any other tools performing the same steps.

```

egonw committed:
Apply suggestions from code review

Co-authored-by: Philippe Rocca-Serra <proccaserra@gmail.com>

egonw committed:
Added reviewer

egonw committed:
Fixed a syntax error introduced in the reviewing process

egonw commented on content/recipes/interoperability/registering-swisslipids-identifiers.md:
```diff
@@ -0,0 +1,171 @@
+:recipe_type: hands_on
+:reading_time_minutes: 15
+:intended_audience: bioinformatician, data_scientist, data_engineer
+:has_executable_code: nope
```
fixme

egonw replied:
lol: https://github.com/FAIRplus/the-fair-cookbook/blob/cdaa3931a60c573b971b5b8be1820820974a78c2/_ext/panels_fairplus.py#L125

egonw committed:
Use the bio.tools link.

egonw committed:
Clarified sentence.

egonw committed:
Updated step 3

- add a note about SHACL (reviewer request)
- a bit more context
- a note that the step is optional

egonw committed:
Added a acknowledgment section

egonw commented:
Moved to https://github.com/FAIRplus/the-fair-cookbook/pull/424
